core/services/ardupilot_manager/mavlink_proxy/Endpoint.py

```diff
@@ -1,76 +1,82 @@
-from dataclasses import dataclass
-from enum import IntEnum
-from typing import Dict, Optional, Union
+from enum import Enum
+from typing import Any, Dict, Optional, Type
 
+import validators
+from pydantic import dataclasses, root_validator
 
-class EndpointType(IntEnum):
-    Undefined = 0
-    UDPServer = 1
-    UDPClient = 2
-    TCP = 3
-    Serial = 4
-    File = 5
 
+def is_valid_filepath(filepath: str) -> bool:
+    return filepath.startswith("/") and not filepath.endswith("/")
```

Comment:
We should save the fine in a predefine, having the filename as a global thing may be a bad idea, since this application will run as root, the user may overwrite important files for the operating system or any other file in the companion!
We should only accept filenames and save the files in a specific folder like `Documents/companion/ArduPilotManager/file_endpoints/file`

Reply:
Agreed!

Reply:
One question that I make myself, though, is "where should this be enforced?"

From my point of view, we can enforce in the Endpoint class itself, making in a way that the file endpoint should always have just a filename, not a complete path, and the ArduPilotManager takes care of completing the path on its way.

Also, for organization purposes, I'm thinking about storing just the endpoint filename in the settings file, so that ArduPilotManager always composes the complete path when loading the files (not when storing).

What do you think?